### PR TITLE
[draft] Drop Python 3.7 and allow tests to work in isolation

### PR DESCRIPTION
**Pull request by graingert**

### CODE REVIEW

**Comment by graingert** [comment]
for the last remaining tests it looks like someone is leaving a ThreadPoolExecutor running and not shutting it down

**Comment by guillaumeeb** [comment]
Thanks @graingert for all the contribution here. I think I'll need some time to digest all this.

I'm a bit confused by the modification for the running loop. Outside of test environment, the current code of dask-jobqueue seems to be working well with recent versions of distributed. And I didn't encounter these errors in my tests, how would you explain it ?

File: dask_jobqueue/core.py [hunk 1]
```diff
@@ -94,7 +96,22 @@
 """.strip()
 
 
-class Job(ProcessInterface, abc.ABC):
+class FixedProcessInterface(ProcessInterface):
```
[the remark this answers is not in the document]
Comment (guillaumeeb) [reply dask_jobqueue/core.py:99]: OK, so from what I'm aware of, we never get into loop questionning on dask-jobqueue. I guess we only use the default loop from MainThread when using it.

From what I understand, this is the `clean` function of `distributed.tests_util` that just removes the default created loop when we use the `loop` fixture. Is that normal that this function might wipe the MainThread loop?

See the last commit of my branch https://github.com/dask/dask-jobqueue/pull/562/commits/d50cc2d0a9a239ebdcf0872f886fc6c30fad6b58 (sorry I did not cherry pick some of your modifications, I still lack git knowledge on top of asyncio stuf).

Maybe this is meaning that we are not doing things right, but might this be a temporary workaround until a bigger refactor?

[the remark this answers is not in the document]
Comment (graingert) [reply dask_jobqueue/core.py:99]: Yes the default event loop is deprecated, and already removed when used with asyncio.run. I added this to our cleanup to make sure we're ready for the removal and support running with multiple calls to asyncio.run:

```python
import asyncio

import distributed

async def noop():
    pass

asyncio.run(noop())

import dask_jobqueue
# create a cluster
RuntimeError: There is no current event loop in thread 'MainThread'
```

[the remark this answers is not in the document]
Comment (guillaumeeb) [reply dask_jobqueue/core.py:99]: Thanks for the clarification. If I understand correctly, at one point dask-jobqueue would have to be ready for the removal of the default event loop, is that right? When will it be supressed?

In the meantime, do you think it's reasonable to use the workaround I proposed, and in parallel open an issue about this problem and try to fix it later on?

[the remark this answers is not in the document]
Comment (graingert) [reply dask_jobqueue/core.py:99]: It's deprecated in 3.10 and scheduled for removal in 3.12

[the remark this answers is not in the document]
Comment (graingert) [reply dask_jobqueue/core.py:99]: I have an idea for a workaround that I want to try tomorrow

[the remark this answers is not in the document]
Comment (graingert) [reply dask_jobqueue/core.py:99]: > If I understand correctly, at one point dask-jobqueue would have to be ready for the removal of the default event loop

The default event loop already gets removed when anyone uses `asyncio.run` and so is already a bug from python 3.6+

[the remark this answers is not in the document]
Comment (graingert) [reply dask_jobqueue/core.py]: my workaround didn't work, so I've made an issue upstream https://github.com/dask/distributed/issues/6858

**Pull request closed without merging**